### ISSUE TITLE
Redesign: Have hunting dwarf collect meat

### ISSUE THREAD

**Issue by com1clyf3** [CompletelyFairGames/dwarfcorp#791]
Currently, after an animal has been hunted, the hunting dwarf will path back to HQ without gathering meat resource. Then another dwarf will path to meat to collect it. Let's make the current hunting dwarf check for meat resource and gather if there's something there.

27b0c862

**Comment by mklingen** [comment]
That's simply not true. Dwarfs that hunt do in fact collect meat from hunted animals.

**Comment by com1clyf3** [comment]
They don't. Take a look at it in game.  If the logic is there, then it's not working for some reason.

**Comment by mklingen** [comment]
Cannot reproduce. Hunting dwarves gather meat. What are the specific steps taken to reproduce?

**Comment by com1clyf3** [comment]
Regressing. This is still occurring on 1fb82458

1. Select animal to hunt.
2. Have dwarves kill it.
3. Observer original killer does not collect the meat resource, but paths away.
4. Another dwarf paths to resource to collect it.

This happens every time for me.

**Comment by mklingen** [comment]
The following happens every time for me, I think there's something more to it than you think:

1. Select animal to hunt.
2. Dwarf kills animal.
3. Same dwarf collects animal meat.

Are you selecting **multiple** animals by chance? If that's the case killing the next animal is higher priority than gathering the meat, in which case it's working as designed.

**Comment by com1clyf3** [comment]
Hm. That seems to be  mostly it. The end result is pretty much always that the last rabbit gets collected by a dwarf that's far away from it.

**Comment by Blecki** [comment]
I'm still seeing the occasional item left behind. Not just from hunting,
also from harvesting plants.

**Comment by com1clyf3** [comment]
If there's a quick fix to that, sure. If it takes more than a day, I say let's not worry abut it.

**Comment by mklingen** [comment]
@Blecki do you want to try this?

**Comment by Blecki** [comment]
Not really. It seems to work 99% of the time still, so I'd consider it a
bug. I think the dwarf is just forgetting the task.
